### ISSUE TITLE
Liquidity is calculated wrong

### ISSUE THREAD

**Issue by amin3378** [gas111-bot/gaspump-sdk#2]
I analyzed jetton master addresses on your platform and the amount of ton accumulated there is far more than the trading fees, which are 0.3 ton per sell, even if you consider 0.3 trading fee for buys it is still a lot more, there is just 9000 tons for example in this address:
https://tonviewer.com/EQDD21KQbeC4wIKChGisYQ1_LHRED7qfT6OMMoUESngAd3ne
you can see even after launching on dedust there is 96 tons left on the jetton master address,  the fees are seen when selling which is around 30 times --> 9 tons, even if you consider buys and sells together it would be around 100 txs --> 30 ton
these funds are accumulated because there is certainly a problem in your algorithm calculating the remaining liquidity which you can see on every coin for example:
https://tonviewer.com/EQBWqEJJGyeKBQl3x1VtIJAnrwVvnHFUAhiLoymlEbfNzcKj
you can see 7.5 tons left on the address, 18 sells --> 6 tons --> actuall liquidty=1.5 tons, but in the app it says 0.93 ton liquidity left
I contacted support multiple times and they said these are fees which clearly are not

**Comment by amin3378** [comment]
i the file GasPumpJetton.ts it is clearly stated that there is only a 0.3 fee for sell which you stated most of it will be refunded

**Comment by karfly** [comment]
Hi!
Thank you for a detailed question with examples.

My initial guess is that some small amount of TON gas is accumulated on contract in addition to commissions. We'll check what really happens and why

**Comment by amin3378** [comment]
Yes if there is a problem it can be critical, please take this in account carefully, I have an excel sheet extracted that I can send you of 10000 tokens like these with total amount of 9000 ton

**Comment by karfly** [comment]
I missed one thing. When token is launched on DeDust we take fixed 50 TON commission (just like pump.fun) and you can see it on the contract as addition to balance.

**Comment by amin3378** [comment]
most of the tokens i'm talking about are tokens that never made it to dedust and most of them everybody sold, but the difference in the wallet balance is seen there too and it's a lot

**Comment by amin3378** [comment]
maybe a portion of this 50 ton comission you are talking about is applied every time somebody buys and sells, this 50 ton should be taken only after the token launched on dedust.

**Comment by amin3378** [comment]
I sent the excel file to your email

**Comment by amin3378** [comment]
Hi any updates on the matter?

**Comment by ggauzin** [comment]
uhm bro they take commitions from those .3 ton

**Comment by amin3378** [comment]
This is not about 0.3 TON, this is about funds taken more than that

**Comment by ggauzin** [comment]
i didn't really get it then

**Comment by Madesupra27** [comment]
Thank so much brother,i'm very support all produk and all project..if me wrong I so very sorry ...please help me so sorry..🧯🙏🙏👍

**Comment by Madesupra27** [comment]
UQBOwGb9Nr55ev-ewf28h7ctRpJnmenJ7AF05U1MuqJ7FGdi, my ton wallet...please help me comision..because i want to make gas pump meme coin..i don't have coin to make..

**Comment by ggauzin** [comment]
get the fuck outta here
